### ISSUE TITLE
Cannot create action with a name that collides with a Python library

### ISSUE THREAD

**Issue by jsjeannotte** [StackStorm/st2#2650]
We created an action called "elasticsearch.py" (with the appropriate yaml). When we try to run it we get an ST2 exception. We are able to reproduce with another action called "test.py". Our hypothesis is that if the action name is the same as a Python library, it fails to run it. This happens after migrating to 1.4.0 from 0.13.2 (it works fine in 0.13.2).

Here's the stacktrace:

	$ st2 run netflix_samples.elasticsearch app_region=us-east-1
	.
	id: 57211d9b7279e6326f6b025f
	status: failed
	parameters: 
	 app_region: us-east-1
	result: 
	 exit_code: 1
	 result: null
	 stderr: "Traceback (most recent call last):
	 File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2actions/runners/python_action_wrapper.py", line 164, in <module>
	   obj.run()
	 File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2actions/runners/python_action_wrapper.py", line 102, in run
	   action = self._get_action_instance()
	 File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2actions/runners/python_action_wrapper.py", line 116, in _get_action_instance
	   actions_cls = action_loader.register_plugin(Action, self._file_path)
	 File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/util/loader.py", line 163, in register_plugin
	   (plugin_abs_file_path))
	Exception: Found no classes in plugin file "/opt/stackstorm/packs/netflix_samples/actions/elasticsearch.py" matching requirements.
	"
	 stdout: ''

**Comment by murgurgula** [comment]
Same issue with  pack names
Example: can't have a pack named datadog if did "pip install datadog"

**Comment by manasdk** [comment]
@jsjeannotte Can you share contents of `test.py` and `test.yaml` where you see this issue?

**Comment by lakshmi-kannan** [comment]
https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/loader.py#L55

This is because of how we use some python module loading  hackery to import classes with actions in them. I can point you to the code but fixing it would require us to move to stevedore like entry_point approach which then makes the whole pack story horrendous IMO.

**Comment by manasdk** [comment]
Suspect it might originate https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/loader.py#L130.

We might need to pass in the pack_name as per https://docs.python.org/2/library/importlib.html#importlib.import_module although there is no guarantee of a package being present from the python perspective. This might be in @Kami wheelhouse.

**Comment by skarmark** [comment]
Any updates on this issue?

**Comment by lakshmi-kannan** [comment]
@skarmark Since there are work arounds for this and really it seems like a trivial problem to me, can we wait on this?

**Comment by skarmark** [comment]
We can wait on this. It is not super urgent but problematic nonetheless. We give freedom to our consumers to select their action and action file names and we have to catch these errors at runtime and ask them to change the action name which is not a very good user experience. Also, changing the action name which is deployed and possibly running is tricky.

**Comment by Kami** [comment]
I will look into it some time in the next couple of days and see if it's possible to do an easy fix.

**Comment by Kami** [comment]
After looking into the issue again more in-depth I don't think we should change this behavior - right now it behaves the same as any other code in Python.

In regular Python there are some ways around it (e.g. using from future import absolute_import), but in our case only possible work-around to get it to work would be to to manipulate `PYTHONPATH` for the Python action sub-process, but this would cause problems since it would mask other imports.

Another option would be to make action/ directory a package (this would require user to add __init__.py file to the directory), but I'm not a big fan of this approach since actions are not necessary Python only.

In any case, I will still play around with it a bit and see if we can come up with a compromise - perhaps make a package thing optional and if __init__.py is not present, it will behave as it does right now.

**Comment by Kami** [comment]
@skarmark #2870 should address it, but I still need to do more testing to make sure it doesn't break or negatively change the way things currently work.

**Comment by Kami** [comment]
@skarmark #2870 has been merged into master. If you feel adventurous, you could try new v1.7dev packages once they are built.

**Comment by skarmark** [comment]
Thank you!!
